applications/GeoMechanicsApplication/custom_utilities/transport_equation_utilities.hpp

```diff
@@ -49,5 +50,26 @@ class GeoTransportEquationUtilities
                RelativePermeability * PermeabilityUpdateFactor * IntegrationCoefficient;
     }
 
+    template <unsigned int TNumNodes>
+    static inline BoundedMatrix<double, TNumNodes, TNumNodes> CalculateCompressibilityMatrix(
+        const Vector&         rNp,
+        double                BiotModulusInverse,
+        double                IntegrationCoefficient,
+        std::optional<double> DtPressureCoefficient = std::nullopt)
+    {
+        return CalculateCompressibilityMatrix(rNp, BiotModulusInverse, IntegrationCoefficient, DtPressureCoefficient);
+    }
+
+    static inline Matrix CalculateCompressibilityMatrix(const Vector& rNp,
+                                                        double        BiotModulusInverse,
+                                                        double        IntegrationCoefficient,
+                                                        std::optional<double> DtPressureCoefficient = std::nullopt)
+    {
+        if (DtPressureCoefficient.has_value())
+            return -PORE_PRESSURE_SIGN_FACTOR * DtPressureCoefficient.value() * BiotModulusInverse *
+                   outer_prod(rNp, rNp) * IntegrationCoefficient;
+        else
+            return -PORE_PRESSURE_SIGN_FACTOR * BiotModulusInverse * outer_prod(rNp, rNp) * IntegrationCoefficient;
```

Comment:
To not repeat part of the equation we could do this part like this:
```suggestion
        auto result = -PORE_PRESSURE_SIGN_FACTOR * BiotModulusInverse * outer_prod(rNp, rNp) * IntegrationCoefficient;
        if (DtPressureCoefficient.has_value()) result *= DtPressureCoefficient.value();
        return result;
```

Reply:
Changed but
1. ublas does not allow *=
2. weird clang formatting happened.

Reply:
Follow Anne's suggestion, DtPressureCoefiicient has been removed from this function.